pokete.py

```diff
@@ -1589,72 +1642,85 @@ def fight_ap_potion(obj, enem, info):
+class Extra_Actions:
 
-def playmap_4_extra_action():
-    playmap_water_extra_action(playmap_4.lake_1.obs)
+    @staticmethod
+    def water(obs):
+        if settings.animations and colors:
+            for obj in obs:
+                if random.randint(0, 9) == 0:
+                    if " " not in obj.char:
+                        obj.rechar([i for i in
+                                    [Color.lightblue+"~"+Color.reset,
+                                        Color.blue+"~"+Color.reset]
+                                        if i != obj.char][0])
+                        if obj.x == figure.x and obj.y == figure.y:
+                            figure.redraw()
 
-def playmap_11_extra_action():
-    playmap_water_extra_action(playmap_11.lake_1.obs)
+    @staticmethod
+    def playmap_4():
+        Extra_Actions.water(ob_maps["playmap_4"].lake_1.obs)
 
-def playmap_18_extra_action():
-    playmap_water_extra_action(playmap_18.lake_1.obs)
+    @staticmethod
+    def playmap_11():
+        Extra_Actions.water(ob_maps["playmap_11"].lake_1.obs)
 
-def playmap_21_extra_action():
-    playmap_water_extra_action(playmap_21.lake_1.obs)
+    @staticmethod
+    def playmap_18():
+        Extra_Actions.water(ob_maps["playmap_18"].lake_1.obs)
 
-def playmap_7_extra_action():
-    for obj in playmap_7.inner_walls.obs + playmap_7.trainers + [eval("playmap_7."+i) for i in map_data["playmap_7"]["balls"] if "playmap_7."+i not in used_npcs or not save_trainers]:
-        if obj.added and math.sqrt((obj.y-figure.y)**2+(obj.x-figure.x)**2) <= 3:
-            obj.rechar(obj.bchar)
-        else:
-            obj.rechar(" ")
+    @staticmethod
+    def playmap_21():
+        Extra_Actions.water(ob_maps["playmap_21"].lake_1.obs)
+
+    @staticmethod
+    def playmap_7():
+        for obj in ob_maps["playmap_7"].inner_walls.obs + ob_maps["playmap_7"].trainers + [getattr(ob_maps["playmap_7"], i) for i in p_data.map_data["playmap_7"]["balls"] if "playmap_7."+i not in used_npcs or not save_trainers]:
+            if obj.added and math.sqrt((obj.y-figure.y)**2+(obj.x-figure.x)**2) <= 3:
+                obj.rechar(obj.bchar)
+            else:
+                obj.rechar(" ")
 
 # NPC functions
 ###############
 
 def playmap_17_boy():
     if "choka" in [i.identifier for i in figure.pokes[:6]]:
-        movemap_text(playmap_17.boy_1.x, playmap_17.boy_1.y,
+        npc = ob_maps["playmap_17"].boy_1
+        movemap_text(npc.x, npc.y,
                     [" < Oh, cool!", " < You have a Choka!",
                     " < I've never seen one before!", " < Here you go, 200$"])
         if ask_bool(movemap, "Young boy gifted you 200$. Do you want to accept it?"):
             figure.add_money(200)
-        playmap_17.boy_1.will = False
-        used_npcs.append(playmap_17.boy_1.name)
+        npc.will = False
+        used_npcs.append(npc.name)
     else:
-        movemap_text(playmap_17.boy_1.x, playmap_17.boy_1.y,
+        movemap_text(npc.x, npc.y,
```

Comment:
Unresolved reference to variable `npc`

Reply:
@MaFeLP `npc` is defined in line 1689 `npc = ob_maps["playmap_17"].boy_1`.

Reply:
PyCharm says otherwise.

Reply:
Look into the code then.

Reply:
I fixed this